lib/orr_msg.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Structures common to BGP, OSPF and ISIS for BGP Optimal Route Reflection
+ * Copyright (C) 2021 Samsung R&D Institute India - Bangalore.
+ *			Madhurilatha Kuruganti
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; see the file COPYING; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#ifndef _FRR_ORR_MSG_H
+#define _FRR_ORR_MSG_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Library code */
+DEFINE_MTYPE_STATIC(LIB, ORR_MSG_INFO, "ORR Msg info");
+
+/* REVISIT: Need to check if we can use zero length array */
+#define ORR_MAX_PREFIX 100
+
+struct orr_prefix_metric {
+	struct prefix prefix;
+	uint32_t metric;
+};
+
+/* BGP-IGP Register for IGP metric */
+struct orr_igp_metric_reg {
+	bool reg;
+	uint8_t proto;
+	safi_t safi;
+	struct prefix prefix;
+};
+
+/* IGP-BGP message structures */
+struct orr_igp_metric_info {
```

Comment:
So the messaging here looks to be "send everything with every update" - is that correct? How much duplication/replication will there be with that approach? Would it be better to consider incremental updates, with 'add' and 'remove' messages?

Reply:
Sure. will modify this to send only modified entries.